### ISSUE TITLE
Why tokenizer pad to right?

### ISSUE THREAD

**Issue by huaiyizhao** [QwenLM/Qwen3-VL#1335]
In the training script, the tokenizer is explicitly converted to right padding, but most existing model use left padding, is there a special reason for that? 
```
tokenizer = transformers.AutoTokenizer.from_pretrained(
        model_args.model_name_or_path,
        cache_dir=training_args.cache_dir,
        model_max_length=training_args.model_max_length,
        padding_side="right",
        use_fast=False,
    )
```

**Comment by JJJYmmm** [comment]
At inference time, we use left padding to avoid next token prediction from the pad token. But it doesn't matter which side to pad in training. Right padding may be more helpful when calculating position ids.

**Comment by huaiyizhao** [comment]
Thanks for the reply. I am still not clear how padding side affects the results, both in training and inference. E.g., I cannot see why padding side doesn't matter in training. The model may see several pad tokens at sentence end, does this affect the model forward pass? Any sources/detailed explainations can I refer to?

**Comment by JJJYmmm** [comment]
In training, "teach forcing" use attention_mask to avoid common tokens to attend to them, pad tokens are also excluded when calculating token loss. For more details, you can refer to https://discuss.huggingface.co/t/the-effect-of-padding-side/67188

**Comment by huaiyizhao** [comment]
I now see the 'magic' is in attention_mask. However, i do not believe explicitly setting padding_side has any effect since in DataCollatorForSupervisedDataset, torch.nn.utils.rnn.pad_sequence is used to pad a batch where the default value of pad_side is 'right' already. What's more counterintuitive is that FlattenedDataCollatorForSupervisedDataset has nothing to do with padding_side at all and this is what i believe the default setting to the training scripts.
